examples_tests/11.RayCastCollision/main.cpp

```diff
@@ -1,159 +1,482 @@
-	collData.attachedNode = sphere;
-	compound->setColliderData(collData);
-    gCollEng->addCompoundCollider(compound);
-    compound->drop();
-
-	uint64_t lastFPSTime = 0;
-
-    core::SColliderData hitPointData;
-	while(device->run() && receiver.keepOpen() )
+	auto cubeTexture = *(gpuImageViews->begin() + EAM_CUBE);
+	auto sphereTexture = *(gpuImageViews->begin() + EAM_SPHERE);
+
+	auto gpuMeshesWireFrame = { createAndGetUsefullData(cubeGeometry, cubeTexture, true), createAndGetUsefullData(sphereGeometry, sphereTexture, true) };
+	auto gpuMeshesNoWireFrame = { createAndGetUsefullData(cubeGeometry, cubeTexture, false), createAndGetUsefullData(sphereGeometry, sphereTexture, false) };
+
+	auto gCollEng = _IRR_NEW(core::SCollisionEngine);
+	auto compound = core::make_smart_refctd_ptr<core::SCompoundCollider>();
+	core::SColliderData hitPointData;
+
+	/*
+		Hot loop for rendering a scene.
+	*/
+
+	while (device->run() && receiver.keepOpen())
 	{
-		driver->beginScene(true, true, video::SColor(255,0,0,255) );
+		driver->beginScene(true, true, video::SColor(255, 255, 255, 255));
 
-        //! This animates (moves) the camera and sets the transforms
-        //! Also draws the meshbuffer
-        smgr->drawAll();
+		camera->OnAnimate(std::chrono::duration_cast<std::chrono::milliseconds>(device->getTimer()->getTime()).count());
+		camera->render();
 
-		driver->endScene();
+		const auto viewProjection = camera->getConcatenatedMatrix();
 
-		setWireframeOnAllMaterials(cube->getMesh(),false);
-		setWireframeOnAllMaterials(sphere->getMesh(),false);
-
-        core::vectorSIMDf origin,dir;
-        origin.set(camera->getAbsolutePosition());
-        dir.set(camera->getTarget());
-        dir -= origin;
-        dir = core::normalize(dir);
-        float outLen;
-        if (gCollEng->FastCollide(hitPointData,outLen,origin,dir,10.f))
-        {
-			if (hitPointData.attachedNode)
+		// there will be switching between wireframing and no wireframing
+		for (auto i = 0; i < EAM_COUNT; ++i)
+		{
+			/*
+				Specific mesh gpu data
+			*/
+
+			auto meshDataNoWireFraming = *(gpuMeshesNoWireFrame.begin() + i);
+			auto meshDataWireFraming = *(gpuMeshesWireFrame.begin() + i);
+
+			auto gpuMeshBuffer = std::get<0>(meshDataNoWireFraming);
+			auto gpuPipeline = std::get<1>(meshDataNoWireFraming);
+			auto gpuPipelineForWireFraming = std::get<1>(meshDataWireFraming);
+			auto gpuubo = std::get<2>(meshDataNoWireFraming);
+			auto gpuDescriptorSet0 = std::get<3>(meshDataNoWireFraming);
+			auto gpuDescriptorSet1 = std::get<4>(meshDataNoWireFraming);
+
+			IGPUDescriptorSet* gpuDescriptorSets[] = { gpuDescriptorSet0.get(), gpuDescriptorSet1.get() };
+
+			/*
+				Handling colider to specify which mode should be turned on
+			*/
+
+			compound = core::make_smart_refctd_ptr<core::SCompoundCollider>();
+			core::SColliderData collData;
+			collData.attachedNode = gpuMeshBuffer.get(); // how to transform GPU buffer pointer to ISceneNode pointer?

```

Comment:
the example is done, pipeline will be switching between themselves but we need to find a way to attach it to the node for collision detection! Needs doing **scene** stuff